### DISCUSSION TITLE
Disabling Freeform's JavaScript and not adding it manually

### DISCUSSION THREAD

**Discussion by thisisjamessmith** [solspace/craft-freeform#477] [Questions & Answers] [answered]
Hiya,

I do not want to allow Freeform to inject its JavaScript. I understand you have an option for "None (add manually)" in your Freeform Script Insertion Location field and I've seen the documentation at https://docs.solspace.com/craft/freeform/v3/developer/js-plugin.html#loading-freeform-js-manually -- but the field is worded in such a way that it sounds like your JS is still required.
My question is whether the basic form functionality will work correctly _without_ loading any of your JS. I understand that your JS provides some extra functionalities, but if I do not use any of those, will everything still work?

(Just for the sake of completeness I am also using nginx static caching and refreshing the CSRF tokens and Freeform formHashes via ajax on page load - not sure if this has any bearing on the question though).

**Comment by gustavs-gutmanis** [comment, marked as answer]
Hi,

The form should work out of the box, but you will have to disable the javascript honeypot for it to work (if you have it enabled).

Some functionality won't work, like rules based hiding/showing of fields, AJAX, table fields, signature fields, phone field patterns, etc.

**Reply by thisisjamessmith** [reply]
Thanks for confirming. I've been using Freeform since its very early EE days like 14 years ago, and I'm sure you'll agree that it has exploded in complexity since then! We build a lot of sites and almost never use any of Freeform's advanced functionality (ajax submissions we prefer to handle ourselves). I understand that you have Express Forms too for simpler use cases, which I like a lot, but there is a still a huge gap between Express Forms' simplicity and Freeform's complexity. (If Express Forms allowed editors to create multi options fields (radios/checkboxes) it would likely satisfy 80% of our needs, but I believe even the Pro version doesn't have that functionality?).

**Reply by kjmartens** [reply]
Thank you for your feedback and your long-time business @thisisjamessmith! 🙂 

You are correct that Express Forms does not have the ability to predefine options in radios/checkboxes at this time. We are considering making a larger update to the Pro edition to enable many more common features such as this in the future. 🙂
